Example/Example/CharacteristicsController.swift

```diff
@@ -49,16 +51,28 @@ class CharacteristicsController: UIViewController {
             }).addDisposableTo(disposeBag)
     }
 
-    private func showWriteFieldForCharacteristic(characteristic: Characteristic) {
+    fileprivate func showWriteFieldForCharacteristic(characteristic: Characteristic) {
         let valueWriteController = UIAlertController(title: "Write value", message: "Specify value in HEX to write ",
                                                      preferredStyle: .alert)
         valueWriteController.addTextField { textField in
-
+            textField.delegate = self
         }
         valueWriteController.addAction(UIAlertAction(title: "Cancel", style: .cancel, handler: nil))
         valueWriteController.addAction(UIAlertAction(title: "Write", style: .default) { _ in
-            print("")
+            
+            self.writeValueForCharacteristic(data: self.Data!, characteristic: characteristic)
+            
             })
+        self.present(valueWriteController, animated: true, completion: nil)
+    }
+    
+    fileprivate func writeValueForCharacteristic(data: String,characteristic: Characteristic) {
+        let _data: NSData = data.dataFromHexadecimalString()!
```

Comment:
I think that we shouldn't call it `_data`. As far as I know it's not common in Swift. 
Maybe let's do a little bit of a rename: `data` -> `hexadecimalString`. Also, extension on string with this name is not clearest options - i would stick maybe to the static function on `Data` like:
```
extension Data {
static func fromHexString(string: String) -> Data {impl...}
}
```
What do you think? Also - let's avoid unwrapping and printing to the console